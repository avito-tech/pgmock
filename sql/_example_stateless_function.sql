begin;

create or replace function public.universal_answer()
    returns integer immutable language sql as
$func$
    select 42;
$func$;

select pgmock.mock($$"${'public.universal_answer'::regproc}"$$);

select pg_temp.setup();

select pg_temp.universal_answer();

select pg_temp.teardown();

select pgmock.mock($$
    {
        "oid": "${'public.universal_answer'::regproc}",
        "mock_name": "another_universal_answer"
    }
$$);

select pg_temp.setup();

select pg_temp.another_universal_answer();

select pg_temp.teardown();

rollback;