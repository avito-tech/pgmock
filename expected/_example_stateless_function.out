begin;
create or replace function public.universal_answer()
    returns integer immutable language sql as
$func$
    select 42;
$func$;
select pgmock.mock($$"${'public.universal_answer'::regproc}"$$);
 mock 
------
 
(1 row)

select pg_temp.setup();
 setup 
-------
 
(1 row)

select pg_temp.universal_answer();
 universal_answer 
------------------
               42
(1 row)

select pg_temp.teardown();
 teardown 
----------
 
(1 row)

select pgmock.mock($$
    {
        "oid": "${'public.universal_answer'::regproc}",
        "mock_name": "another_universal_answer"
    }
$$);
 mock 
------
 
(1 row)

select pg_temp.setup();
 setup 
-------
 
(1 row)

select pg_temp.another_universal_answer();
 another_universal_answer 
--------------------------
                       42
(1 row)

select pg_temp.teardown();
 teardown 
----------
 
(1 row)

rollback;
